integration/runner/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+const cadvisorBinary = "cadvisor"
+
+func RunCommand(cmd string, args ...string) error {
+	output, err := exec.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("command %q %q failed with error: %v and output: %q", cmd, args, err, output)
+	}
+
+	return nil
+}
+
+func Run() error {
+	start := time.Now()
+	defer func() {
+		glog.Infof("Execution time %v", time.Since(start))
+	}()
+	defer glog.Flush()
+
+	host := flag.Args()[0]
+	testDir := fmt.Sprintf("/tmp/cadvisor-%d", os.Getpid())
+	glog.Infof("Running integration tests in host %q", host)
+
+	// Build cAdvisor.
+	glog.Infof("Building cAdvisor...")
+	err := RunCommand("godep", "go", "build", "github.com/google/cadvisor")
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := RunCommand("rm", cadvisorBinary)
+		if err != nil {
+			glog.Error(err)
+		}
+	}()
+
+	// Ship it to the destination host.
+	glog.Infof("Pushing cAdvisor binary to remote host...")
+	err = RunCommand("gcutil", "ssh", host, "mkdir", "-p", testDir)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := RunCommand("gcutil", "ssh", host, "rm", "-rf", testDir)
+		if err != nil {
+			glog.Error(err)
+		}
+	}()
+	err = RunCommand("gcutil", "push", host, cadvisorBinary, testDir)
+	if err != nil {
+		return err
+	}
+
+	// TODO(vmarmol): Get logs in case of failures.
+	// Start it.
+	glog.Infof("Running cAdvisor on the remote host...")
+	err = RunCommand("gcutil", "ssh", host, "sudo", "screen", "-d", "-m", path.Join(testDir, cadvisorBinary), "--logtostderr", "&>", "/dev/null")
```

Comment:
I thought some of the integration tests assume that cadvisor is running inside a container.